### PR TITLE
1.2.1 tidb-operator-intro

### PR DESCRIPTION
**Pull request by shonge**

### CODE REVIEW

File: session2/chapter1/tidb-operator-intro.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# TiDB Operator 简介及原理
+
+​                             *—— 化繁为简的艺术*
+
+## 一、背景
+
+小时候，想象着2020年机器人早就普及了，机器人管家叫我起床、机器人厨师端上早餐、机器人司机开车送我去上班，路上还有机器人交警，到了公司是机器人姐姐接待，下班去看的是机器人牙医而且一点也不疼。
```
Comment (DanielZhangQD) [inline session2/chapter1/tidb-operator-intro.md]:
```suggestion
小时候，想象着 2020 年机器人早就普及了，机器人管家叫我起床、机器人厨师端上早餐、机器人司机开车送我去上班，路上还有机器人交警，到了公司是机器人姐姐接待，下班去看的是机器人牙医而且一点也不疼。
```

File: session2/chapter1/tidb-operator-intro.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+“快醒醒，客户爸爸问你TiDB集群怎么还没搞好？”
```
Comment (DanielZhangQD) [inline session2/chapter1/tidb-operator-intro.md]:
```suggestion
“快醒醒，客户爸爸问你 TiDB 集群怎么还没搞好？”
```

File: session2/chapter1/tidb-operator-intro.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+“天啊，为什么都2020年了还要人手动去维护分布式集群？为什么这群人要把TiDB设计成这么多组件？为什么TiDB集群半夜挂了不派个机器人去修复？”
```
Comment (DanielZhangQD) [inline session2/chapter1/tidb-operator-intro.md]:
```suggestion
“天啊，为什么都 2020 年了还要人手动去维护分布式集群？为什么这群人要把 TiDB 设计成这么多组件？为什么 TiDB 集群半夜挂了不派个机器人去修复？”
```

File: session2/chapter1/tidb-operator-intro.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+“兄dei，K8s了解一下？“
```
Comment (DanielZhangQD) [inline session2/chapter1/tidb-operator-intro.md]:
```suggestion
“兄dei，K8s 了解一下？“
```

File: session2/chapter1/tidb-operator-intro.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+TiDB Operator 像“牧羊人”一样，持续的监督并管理着 TiDB 各组件“羊群”以恰当的状态运行在主机集群“牧场”上。现在运维人员只要告诉 Operator “What  to do“，而由 Operator来决定 “How to do”。在最新版本 TiDB Operator 甚至可以根据实际情况来决定 "What to do"，比如：auto-scaler。真正实现了自动化运维，减轻运维人员维护压力，提高服务能力。
+
+### TiDB Operator 架构
+
+![TiDB Operator 架构](https://download.pingcap.com/images/docs-cn/stable/tidb-operator-overview.png)
+
+### TiDB Operator 组件
+
+1. TiDB Cluster 定义：CRD（`CustomResourceDefinition`）定义了`tidbcluster`等自定义资源，使得 Kubernetes 世界认识 TiDB Cluster 并让其与 `Deployment`、`Statefulset` 一同享受 Kubernetes 的头等公民待遇。目前TiDB Operator v1.1.0 版本包含的CRD有：`TidbCluster`、`Backup`、`Restore`、`BackupSchedule`、`TidbMonitor`、`TidbInitializer`以及`TidbClusterAutoScaler`。
```
Comment (DanielZhangQD) [inline session2/chapter1/tidb-operator-intro.md]:
```suggestion
1. TiDB Cluster 定义：CRD（`CustomResourceDefinition`）定义了`tidbcluster`等自定义资源，使得 Kubernetes 世界认识 TiDB Cluster 并让其与 `Deployment`、`Statefulset` 一同享受 Kubernetes 的头等公民待遇。目前 TiDB Operator v1.1.0 版本包含的 CRD 有：`TidbCluster`、`Backup`、`Restore`、`BackupSchedule`、`TidbMonitor`、`TidbInitializer` 以及 `TidbClusterAutoScaler`。
```

File: session2/chapter1/tidb-operator-intro.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+2. 控制器：`tidb-controller-manager` 包含了一组自定义控制器，控制器通过循环不断比对被控制对象的期望状态与实际状态，并通过自定义的逻辑驱动被控制对象达到期望状态。
+3. 调度器：`tidb-scheduler` 是一个 Kubernetes 调度器扩展，它为 Kubernetes 调度器注入 TiDB 集群特有的调度逻辑，比如：为保证高可用，任一 Node 不能调度超过 TiDB 集群半数以上的tikv实例。
```
Comment (DanielZhangQD) [inline session2/chapter1/tidb-operator-intro.md]:
```suggestion
3. 调度器：`tidb-scheduler` 是一个 Kubernetes 调度器扩展，它为 Kubernetes 调度器注入 TiDB 集群特有的调度逻辑，比如：为保证高可用，任一 Node 不能调度超过 TiDB 集群半数以上的 tikv 实例。
```

File: session2/chapter1/tidb-operator-intro.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+### 自定义资源
+
+1. TiDB Cluster 资源：CR（`CustomResource`）声明了 TiDB Cluster 自定义资源对象，它声明了`tidbcluster`对象的期望状态，并被控制器逻辑不断进行处理，同时将实际运行状态记录下来。
```
Comment (DanielZhangQD) [inline session2/chapter1/tidb-operator-intro.md]:
```suggestion
1. TiDB Cluster 资源：CR（`CustomResource`）声明了 TiDB Cluster 自定义资源对象，它声明了`tidbcluster` 对象的期望状态，并被控制器逻辑不断进行处理，同时将实际运行状态记录下来。
```

File: session2/chapter1/tidb-operator-intro.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+### Kubernetes 控制平面
+
+1. `Kube-apiserver`：Kubernetes 控制平面的前端，所有组件通过API Server获取或更新对象信息。
```
Comment (DanielZhangQD) [inline session2/chapter1/tidb-operator-intro.md]: ### 
```suggestion
1. `Kube-apiserver`：Kubernetes 控制平面的前端，所有组件通过 API Server 获取或更新对象信息。
```

File: session2/chapter1/tidb-operator-intro.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+2. `Kube-controller-manager`：TiDB Cluster 等CRD封装了`StatefulSet`、`Deployment`、`CronJob`等原生对象，所以依然需要K8s原生控制器来进行控制逻辑。
```
Comment (DanielZhangQD) [inline session2/chapter1/tidb-operator-intro.md]:
```suggestion
2. `Kube-controller-manager`：TiDB Cluster 等 CR 封装了`StatefulSet`、`Deployment`、`CronJob` 等原生对象，所以依然需要 K8s 原生控制器来进行控制逻辑。
```

File: session2/chapter1/tidb-operator-intro.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+3. `Kube-scheduler`：调度TiDB Cluster等CRD所运行的Pod，需要先经过`tidb-scheduler`的筛选，然后再使用K8s原生调度器来进行Pod调度。
```
Comment (DanielZhangQD) [inline session2/chapter1/tidb-operator-intro.md]:
```suggestion
3. `Kube-scheduler`：调度 TiDB Cluster 的 Pod，`filtering` 阶段，kube-scheduler 筛选出的节点会再经过 `tidb-scheduler` 筛选一次，然后 kube-scheduler 再进行 `scoring` 选择最合适的节点进行Pod调度。
```

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by cofyc — APPROVED** [review]
LGTM

**Review by tennix — APPROVED** [review]
LGTM

**Pull request merged**